src/as/glas/core/EventDispatcher.spec.ts

```diff
@@ -2,70 +2,139 @@
  * @author simonThiele / https://github.com/simonThiele
  * @author TristanVALCKE / https://github.com/Itee
  * @author corruptedzulu / https://github.com/corruptedzulu
+ * @author Joe Pea / http://github.com/trusktr
  */
 /* global QUnit */
 
-import {EventDispatcher, Listener} from './EventDispatcher'
+import {EventDispatcher, Listener, ListenerArray} from './EventDispatcher'
 import {Event} from './Event'
 
-var callCount: i32 = 0
+let callCount: i32 = 0
 
 describe('EventDispatcher', (): void => {
-	describe('.Instancing', (): void => {
-		test('Instancing', (): void => {
-			// expect<number>(a.x).toBe(0)
-		})
+	test('.constructor', (): void => {
+		// expect<number>(a.x).toBe(0)
 	})
-	describe('.addEventListener', (): void => {
-		test('addEventListener', (): void => {
-			// var eventDispatcher = new EventDispatcher()
-			// var listener: Listener = (event: Event) => void
-			// eventDispatcher.addEventListener('anyType', listener)
-			// expect<number>(eventDispatcher._listeners.anyType.length).toBe(1);
-			// expect<Listener>(eventDispatcher._listeners.anyType[0]).toStrictEqual(listener);
-			// eventDispatcher.addEventListener('anyType', listener)
-			// expect<number>(eventDispatcher._listeners.anyType.length).toBe(1);
-			// expect<Listener>(eventDispatcher._listeners.anyType[0]).toStrictEqual(listener);
-		})
+
+	test('.addEventListener', (): void => {
+		const eventDispatcher = new EventDispatcher()
+		const listener: Listener = (event: Event) => {}
+
+		eventDispatcher.addEventListener('anyType', listener)
+
+		// prettier-ignore
+		expect<ListenerArray>(
+			eventDispatcher
+				// @ts-ignore private access
+				._listeners
+				.get('anyType')
+		).toHaveLength(1);
+
+		// // prettier-ignore
+		// expect<Listener>(
+		// 	eventDispatcher
+		// 		// @ts-ignore private access
+		// 		._listeners
+		// 		.get('anyType')[0]
+		// ).toBe(listener)
```

Comment:
Without these tests enabled, the below `.dispatchEvent` along with this `.addEventListener` test proves that the function must be the same reference.

@jtenner How can we check the listener reference? This commented code gives the error

```
ERROR AS203: Operation 'offsetof' cannot be applied to type '(event: src/as/glas/core/Event/Event) => void'.

         Actual.offset = offsetof<T>();
                                  ~
 in node_modules/@as-pect/assembly/assembly/internal/report/Actual.ts(152,33)

ERROR AS203: Operation 'offsetof' cannot be applied to type '(event: src/as/glas/core/Event/Event) => void'.

         Expected.offset = offsetof<T>();
                                    ~
 in node_modules/@as-pect/assembly/assembly/internal/report/Expected.ts(161,35)

[Error] There was a compilation error when trying to create the wasm binary for file: src/as/glas/core/EventDispatcher.spec.ts.
```

Reply:
This is an unexpected (valid) use case! I will open up a pull request on aspect to fix this on Monday.

Instead of using `.toContain` you can loop over the array yourself for now.

Reply:
@jtenner Ah, thanks! The following works:

```ts
		expect<bool>(
			eventDispatcher
				._listeners
				.get('anyType')[0] == listener
		).toBe(true)
```